beetsplug/discogs.py

```diff
@@ -205,11 +205,27 @@ def get_albums(self, query):
             self._log.debug(u"Communication error while searching for {0!r}",
                             query, exc_info=True)
             return []
-        return [self.get_album_info(release) for release in releases[:5]]
+        return filter(None, [self.get_album_info(release)
+                             for release in releases[:5]])
```

Comment:
Noted - I was mainly trying not to end up with a too cumbersome couple of lines (as the `if x` part would involve calling the function again, etc). I have replaced it with a version that is slightly convoluted but returns a list, which seems the right thing to do as it passed upwards to the mechanism that deal with `candidates`.